community/graphdb-api/src/main/java/org/neo4j/graphdb/security/WriteOperationsNotAllowedException.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.graphdb.security;
+
+import org.neo4j.kernel.api.exceptions.Status;
+
+public class WriteOperationsNotAllowedException extends RuntimeException implements Status.HasStatus
+{
+    private Status statusCode = Status.General.DatabaseUnavailable;
```

Comment:
Hmm yeh, I'm not sure how we can make it clear why the write was rejected. In the HA/Community modes the reason would be that you had a 'read_only' database but in Core-Edge mode it'd be because you weren't leader which isn't a read-only database so it might be confusing to see that in the logs.